src/Input/Textarea/Textarea.tsx

```diff
@@ -9,133 +9,116 @@ import {
-      e.target.rows = maxRows;
-      e.target.scrollTop = e.target.scrollHeight;
+    if (textarea.value.length > 0) {
+      setFloating(true);
     }
 
-    this.setState({
-      rows: currentRows < maxRows ? currentRows : maxRows,
-    });
+    setTextareaMaxHeight(~~(textarea.offsetHeight * 3.7 + 23 * 8 + 1));
+  }, []);
 
-    if (isFunction(this.props.onChange)) {
-      return this.props.onChange(e);
-    }
-  };
+  const handleChange = React.useCallback(
+    (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+      const previousRows = e.target.rows;
+      e.target.rows = minRows;
 
-  componentDidMount() {
-    const textarea = this.textareaInputRef.current;
+      const currentRows = ~~(e.target.scrollHeight / 30);
 
-    if (textarea.value.length > 0) {
-      this.setState({
-        floating: true,
-      });
-    }
+      if (currentRows === previousRows) {
+        e.target.rows = currentRows;
+      }
 
-    this.setState({
-      textareaMaxHeight: ~~(textarea.offsetHeight * 3.7 + 23 * 8 + 1),
-    });
-  }
+      if (currentRows >= maxRows) {
+        e.target.rows = maxRows;
+        e.target.scrollTop = e.target.scrollHeight;
+      }
 
-  componentDidUpdate() {
-    const { value } = this.props;
-    const { floating } = this.state;
+      setRows(currentRows < maxRows ? currentRows : maxRows);
 
+      if (isFunction(onChange)) {
+        return onChange(e);
+      }
+    },
+    [onChange]
+  );
+
+  const handleFocusChange = React.useCallback(
+    (onBlur: (e: React.FocusEvent<HTMLTextAreaElement>) => void) => {
+      const listener = (e: React.FocusEvent<HTMLTextAreaElement>) => {
+        setFloating(e.target.value.length > 0);
+
+        if (onBlur !== undefined) {
+          return onBlur(e);
+        }
+      };
+
+      return listener;
+    },
+    []
+  );
+
+  React.useEffect(() => {
     if (value && value !== '' && !floating) {
-      this.setState({ floating: true });
+      setFloating(true);
     }
-  }
-
-  render() {
-    const {
-      label,
-      value,
-      status,
-      disabled,
-      className,
-      onBlur,
-      onChange,
-      removeFloatingLabel,
-      ...restProps
-    } = this.props;
-
-    const { floating, rows, textareaMaxHeight } = this.state;
-
-    return (
-      <TextareaContainer className={classNames('aries-textarea', className)}>
-        <TextareaInput
-          ref={this.textareaInputRef}
-          placeholder={removeFloatingLabel && label}
-          rows={rows}
-          status={status}
-          disabled={disabled}
-          onBlur={this.handleFocusChange(onBlur)}
-          onChange={this.handleChange}
+  }, [value, floating]);
+
+  return (
+    <TextareaContainer className={classNames('aries-textarea', className)}>
+      <TextareaInput
+        ref={textareaInputRef}
+        placeholder={removeFloatingLabel && label}
+        rows={rows}
+        status={status}
+        disabled={disabled}
+        onBlur={handleFocusChange(onBlur)}
```

Comment:
Won't it break `useCallback`? Might need to refactor it to be like:
```
  const handleFocusChange = React.useCallback(
    (e: React.FocusEvent<HTMLTextAreaElement>) => {
      setFloating(e.target.value.length > 0);

      if (onBlur !== undefined) {
        return onBlur(e);
      }
    },
    [onBlur]
  );
```
```

Reply:
I added `onBlur` as a dependency for the `useCallback` but I couldn't get your code to work. It complains about type errors and I don't know how to fix that. What exactly is your concern here? It seems to be working the way it is.

Reply:
Oh, I think you need to do this https://github.com/glints-dev/glints-aries/pull/238#discussion_r384371313 to fix that type error.

Reply:
My concern is `onBlur={handleFocusChange(onBlur)}` is actually recreated fucntion in every render, we can refactor `handleFocusChange` to make it be `onBlur={handleFocusChange}` which should improve performance.

Reply:
Updated!